### PR TITLE
Updates for release 0.28.0 development

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** This PR prepare the package for the 0.28 release cycle. The following changes will be added:
- Update release tag to 0.28.0_devx
- Pin CMake to 3.24.x to avoid pitfalls encountered with 3.25
- Deprecate Python 3.7 support

**Description of the Change:** As above.

**Benefits:** Package updates, continued builds, and reduced build overhead.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
Thank you for that!

File: .github/workflows/build_wheel_manylinux2014.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
   pull_request:
 
 env:
-  CIBW_BUILD: 'cp37-* cp38-* cp39-* cp310-* cp311-*'
+  CIBW_BUILD: 'cp38-* cp39-* cp310-* cp311-*'
```
Comment (AmintorDusko) [inline .github/workflows/build_wheel_manylinux2014.yml:10]: Nice!

**Review by multiphaseCFD — APPROVED** [review]
Nice work @mlxd ! No more comments.

**Comment by mlxd** [comment]
Thanks folks!

**Pull request merged**